### ISSUE TITLE
Maps (including exported heatmaps) show "This page cannot load Google Maps correctly"

### ISSUE THREAD

**Issue by bartvanandel** [GoldenCheetah/GoldenCheetah#2901]
The map view, as well as exported heatmaps, started showing a message saying "This page cannot load Google Maps correctly". This has to do with a missing API key when loading the Google Maps API (the javascript libraries), which is also mentioned in the Dev console.

Pointer: https://developers.google.com/maps/documentation/javascript/error-messages#no-api-keys

Possible solutions:
- obtain API key and distribute it with GC
- allow user to add it themselves
- use alternative, such as Leaflet or OpenLayers

**Comment by batsonjay** [comment]
Sorry to hear this, as it removes one of my primary use cases for GC. (Long-time user.)

Is there no (free) alternate map source?

**Comment by liversedge** [comment]
maps chart will remain, using openstreet api, heatmap will go.

**Comment by batsonjay** [comment]
Thanks. Using GC to select route based on past data is key use; but ANY map source is ok (and OSM is good).  Google be damned.

**Comment by bartvanandel** [comment]
If you're interested, I'm experimenting locally with an OpenLayers based version of the heat map. Note that I'm not currently applying this to GC source code (or well, a template file), just modifying the output here. I've got a somewhat working version which I'm still optimizing. If I'm happy with it I can send you an example.

It'd really be a shame if the heatmap were removed, this is actually one of the features I'm most happy about.

**Comment by liversedge** [comment]
If you can patch it, we'll do that. Its not a particularly popular feature and I'm all for simplifying things, but not chopping nose off to spite face !

**Comment by bartvanandel** [comment]
Playing around with Leaflet it seems that one is even faster. Just modified an existing demo to use 100,000 point, no sweat, very promising. I'll see if I can provide a patch once I've got something stable.
Might use the same for the regular map.

**Comment by philippgille** [comment]
@bartvanandel @liversedge: Is there any progress on moving from Google Maps to another provider?

I noticed the same error message in the newest version of GoldenCheetah. If the maps feature isn't supported anymore, maybe it makes sense to release a new version without the tab?

One question by the way: I didn't find where to open the JS console within GoldenCheetah. I was assuming it's an embedded Chromium without the controls or something similar, but the initial creator of this issue seems to have been able to open the console and see more detailed error messages regarding the maps.

And for the archive, in GoldenCheetah 3.4, with language set to German, the error looks like this:

![grafik](https://user-images.githubusercontent.com/170670/66594093-aa865f00-eb98-11e9-848d-4e063aadb862.png)

**Comment by amtriathlon** [comment]
There was a recent discussion on the users forum about this topic: https://groups.google.com/d/msg/golden-cheetah-users/BLd6Qe42RJk/Y57gw8ZJCQAJ
TL;DR: to use maps you need to switch to v3.5
